ScrapPackedLibrary/ScrapPackedFile.cs

```diff
@@ -257,89 +265,132 @@ private void ExtractFile(string p_packedPath, string p_destinationPath, FileStre
                     
                     fsExtractFile.Write(readBytes);
                 }
+                catch (Exception ex)
+                {
+                    RestoreBackup(p_destinationPath + ".bak");
+                    throw ex;
+                }
                 finally
                 {
                     fsExtractFile.Close();
                 }
             }
+            catch (Exception ex)
+            {
+                RestoreBackup(p_destinationPath + ".bak");
+                throw ex;
+            }
             finally
             {
                 if (p_PackedFileStream == null)
                     fsPacked.Close();
-            }            
+            }
+
+            DeleteBackup(p_destinationPath + ".bak");
+        }
+
+        private void MakeBackup(string filePath)
+        {
+            if (filePath.EndsWith(".bak"))
+                throw new ArgumentException($"{filePath} is already a backup");
+
+            if (File.Exists(filePath))
+            {
+                File.Move(filePath, filePath + ".bak", true);
+                if (filePath == fileName)
+                    fileName = filePath + ".bak";
+            }
+        }
+
+        private void RestoreBackup(string filePath)
+        {
+            if (!filePath.EndsWith(".bak"))
+                throw new ArgumentException($"{filePath} is not a backup");
```

Comment:
Check line 271 or 281. It is already works like you want, if I understood you correctly (not counting random names, will implement it).

Reply:
was trying to say that instead of 
`RestoreBackup(filename+".bak")`
we could call
`RestoreBackup(filename)`
and use
`filePath+".bak"` inside `RestoreBackup`
yes it already works correctly … but this way might be nicer …
…
but it might be more difficult if we do the random names

Reply:
Oh, got it